tools/docker-network/tests/eventapi_test.go

```diff
@@ -415,19 +385,51 @@ func test_NFTTransactionBlocks(t *testing.T, d *DockerTestFramework) {
 		}
 
 		// wait until all topics starts listening
-		err = AwaitEventAPITopics(d.optsWaitFor, cancel, finish, totalTopics)
+		err := e.AwaitEventAPITopics(t, cancel, totalTopics)
 		require.NoError(t, err)
 
 		// issue blocks
 		go func() {
 			for _, blk := range expectedBlocks {
 				fmt.Println("submitting a block ", blk.MustID().ToHex())
-				d.SubmitBlock(context.Background(), blk)
+				e.dockerFramework.SubmitBlock(context.Background(), blk)
 			}
 		}()
 
 		// wait until all topics receives all expected objects
-		err = AwaitEventAPITopics(d.optsWaitFor, cancel, finish, totalTopics)
+		err = e.AwaitEventAPITopics(t, cancel, totalTopics)
+		require.NoError(t, err)
+	}
+}
+
+func test_BlockMetadataMatchedCoreAPI(t *testing.T, e *EventAPIDockerTestFramework) {
+	// get event API client ready
+	ctx, cancel := context.WithCancel(context.Background())
+	eventClt := e.ConnectEventAPIClient(ctx)
+	defer eventClt.Close()
+
+	{
+		account := e.dockerFramework.CreateAccount()
+
+		assertions := []func(){
+			func() { e.AssertBlockMetadataStateAcceptedBlocks(ctx, eventClt) },
+			func() { e.AssertBlockMetadataStateConfirmedBlocks(ctx, eventClt) },
+		}
+
+		totalTopics := len(assertions)
+		for _, assertion := range assertions {
+			assertion()
+		}
+
+		// wait until all topics starts listening
+		err := e.AwaitEventAPITopics(t, cancel, totalTopics)
 		require.NoError(t, err)
+
+		// issue blocks
+		e.SubmitDataBlockStream(account, 5*time.Minute)
+
+		// time's up, cancel the context
+		time.Sleep(5 * time.Minute)
```

Comment:
why do we wait here?
Shouldn't the `SubmitDataBlockStream` return earliest after the 5 minutes anyway?

Reply:
Ah, this is for debugging, to let the network run a bit longer so I can check with core API. I'll remove it.